extractor/docker/docker.go

```diff
@@ -186,41 +180,72 @@ func ApplyLayers(layers []types.LayerInfo) types.ImageDetail {
 }
 
 func (d Extractor) ImageName() string {
-	return d.image.Name()
+	return d.imageName
 }
 
-func (d Extractor) ImageID() digest.Digest {
-	return d.image.ConfigInfo().Digest
+func (d Extractor) ImageID() (string, error) {
+	h, err := d.image.ConfigName()
+	if err != nil {
+		return "", xerrors.Errorf("unable to get the image ID: %w", err)
+	}
+	return h.String(), nil
 }
 
-func (d Extractor) ConfigBlob(ctx context.Context) ([]byte, error) {
-	return d.image.ConfigBlob(ctx)
+func (d Extractor) ConfigBlob() ([]byte, error) {
+	return d.image.RawConfigFile()
 }
 
-func (d Extractor) LayerIDs() []string {
-	return d.image.LayerIDs()
+func (d Extractor) LayerIDs() ([]string, error) {
+	conf, err := d.image.ConfigFile()
+	if err != nil {
+		return nil, xerrors.Errorf("unable to get the config file: %w", err)
+	}
+
+	var layerIDs []string
+	for _, d := range conf.RootFS.DiffIDs {
+		layerIDs = append(layerIDs, d.String())
+	}
+	return layerIDs, nil
```

Comment:
These seem to be untested.

Reply:
They are tested through `analyzer_test.go`.